Sources/Private/CoreAnimation/Layers/ShapeLayer.swift

```diff
@@ -414,5 +412,16 @@ extension Array where Element == ShapeItemLayer.Item {
           otherItems: [strokeOrFill] + otherItems)
       }
     }
+
+    var unusedItems = [ShapeItemLayer.Item]()
+    for renderGroup in renderGroups {
+      // All valid render groups must have a path, otherwise the items wouldn't be rendered
+      if renderGroup.pathItems.isEmpty {
+        unusedItems.append(contentsOf: renderGroup.otherItems)
+      }
+    }
+
+    renderGroups.removeAll(where: { $0.pathItems.isEmpty })
```

Comment:
Rather than doing two passes over `renderGroups` could we iterate its indexes in reverse and remove items by index as we find empty ones?

Reply:
Good idea